src/app/Dashboard/DraggableRef.tsx

```diff
@@ -0,0 +1,139 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import { dashboardCardConfigResizeCardIntent } from '@app/Shared/Redux/Configurations/DashboardConfigSlicer';
+import { gridSpans } from '@patternfly/react-core';
+import _ from 'lodash';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { DashboardCardContext } from './ResizableCard';
+
+export interface DraggableRefProps {
+  dashboardIdx: number;
+  minimumSpan: gridSpans;
+}
+
+function normalizeAsGridSpans(val: number, min: number, max: number, a: number, b: number): gridSpans {
+  if (val < min) val = min;
+  else if (val > max) val = max;
+  const ans = Math.round((b - a) * ((val - min) / (max - min)) + a);
+  return _.clamp(ans, a, b) as gridSpans;
+}
+
+export const DraggableRef: React.FunctionComponent<DraggableRefProps> = (props) => {
+  const dispatch = useDispatch();
+
+  const cardRef = React.useContext(DashboardCardContext);
+  const draggableRef = React.useRef<HTMLDivElement>(null);
+  const isResizing = React.useRef<boolean>(false);
+  const setInitialVals = React.useRef<boolean>(true);
+
+  const SMALLEST_CARD_WIDTH = 126;
+
+  const handleOnMouseDown = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    e.stopPropagation();
+    e.preventDefault();
+    document.body.style.setProperty('cursor', 'col-resize');
+    document.addEventListener('mousemove', callbackMouseMove);
+    document.addEventListener('mouseup', callbackMouseUp);
+    isResizing.current = true;
+    setInitialVals.current = true;
+  };
+
+  const handleMouseMove = (e: MouseEvent) => {
+    const mousePos = e.clientX;
+    e.stopPropagation();
+    if (!isResizing.current) {
+      return;
+    }
+    if (cardRef.current && window.visualViewport) {
+      cardRef.current.scrollIntoView({ behavior: 'smooth', block: 'center', inline: 'end' });
+      const cardRect = cardRef.current.getBoundingClientRect();
+      const cardLeft = cardRect.left + SMALLEST_CARD_WIDTH * props.minimumSpan;
+
+      if (setInitialVals.current) {
+        setInitialVals.current = false;
+      }
+      const newSize = mousePos;
+
+      const gridSpan = normalizeAsGridSpans(
+        newSize,
+        cardLeft,
+        window.visualViewport.width,
+        props.minimumSpan,
+        12
+      ) as gridSpans;
+
+      dispatch(dashboardCardConfigResizeCardIntent(props.dashboardIdx, gridSpan));
+    } else {
+      console.error('cardRef.current or window.visualViewport is undefined');
+    }
+  };
+
+  const handleOnMouseUp = () => {
+    if (!isResizing.current) {
+      return;
+    }
+    isResizing.current = false;
+    setInitialVals.current = true;
+    document.body.style.removeProperty('cursor');
+    document.removeEventListener('mousemove', callbackMouseMove);
+    document.removeEventListener('mouseup', callbackMouseUp);
+  };
+
+  const callbackMouseMove = React.useCallback(handleMouseMove, [
```

Comment:
Probably, bring the `handle*` functions right into their corresponding `callback*` function since now, these `handle*` will get generated/assigned in every re-render.